consult.el

```diff
@@ -2334,13 +2334,9 @@ These configuration options are supported:
                                         'consult--completion-candidate-hook)
                      (let ((enable-recursive-minibuffers t))
                        (if (eq category 'file)
-                           ;; When completing files with consult-completion-in-region, the point in the
-                           ;; minibuffer gets placed initially at the beginning of the last path component.
-                           ;; By using the filename as DIR argument (second argument of read-file-name), it
-                           ;; starts at the end of minibuffer contents, as for other types of completion.
-                           ;; However this is undefined behavior since initial does not only contain the
-                           ;; directory, but also the filename.
-                           (read-file-name prompt initial initial require-match nil predicate)
+                           (minibuffer-with-setup-hook
```

Comment:
Two small suggestions:

1. Use consult--minibuffer-with-setup-hook which fixes some issue with minibuffer-with-setup-hook. I do this consistently throughout consult I think.
2. Still keep the comment partially or explain the history of the change? Usually such subtleties are forgotten again after some time.

Reply:
1. Sure, I'll change it. I forgot about that problem. It was a memory leak, right? And your macro uses a symbol, setting its function value, for the hook instead, I believe.
2. In my view, the comment explains a hack and the new code is a more or less direct expression of intent, so it doesn't need a comment. But I can certainly add some form of comment.

Reply:
Let's just add a comment which links to the issue.

Reply:
Oh, sorry, I made some changes before I saw that. I added a comment that explains why it's important to insert the full initial text into the minibuffer. But I still don't like to the issue.